packages/ripple-binary-codec/src/enums/index.ts

```diff
@@ -119,46 +134,141 @@ function buildField([name, info]: [string, FieldInfo]): FieldInstance {
     ordinal: (typeOrdinal << 16) | info.nth,
     type: new Bytes(info.type, typeOrdinal, TYPE_WIDTH),
     header: field,
-    associatedType: SerializedType, // For later assignment in ./types/index.js
+    associatedType: SerializedType, // For later assignment in ./types/index.js or Definitions.updateAll(...)
   }
 }
 
 /*
  * @brief: The collection of all fields as defined in definitions.json
  */
 class FieldLookup {
-  constructor(fields: Array<[string, FieldInfo]>) {
+  constructor(
+    fields: Array<[string, FieldInfo]>,
+    types: Record<string, number>,
+  ) {
     fields.forEach(([k, v]) => {
-      this[k] = buildField([k, v])
-      this[this[k].ordinal.toString()] = this[k]
+      this.add(k, v, types[v.type])
     })
   }
 
+  public add(name: string, field_info: FieldInfo, typeOrdinal: number): void {
+    this[name] = buildField([name, field_info], typeOrdinal)
+    this[this[name].ordinal.toString()] = this[name]
+  }
+
   fromString(value: string): FieldInstance {
     return this[value] as FieldInstance
   }
 }
+/**
+ * Stores the various types and fields for rippled to be used to encode/decode information later on.
+ *
+ */
+class DefinitionContents {
+  field: FieldLookup
+  ledgerEntryType: BytesLookup
+  type: BytesLookup
+  transactionResult: BytesLookup
+  transactionType: BytesLookup
+  transactionNames: string[]
+  addedDataTypes: Record<string, typeof SerializedType>
+
+  /**
+   * Present rippled types in a typed and updatable format.
+   * For an example of the input format see `definitions.json`
+   * To generate a new definitions file from rippled source code, use this tool: https://github.com/RichardAH/xrpl-codec-gen
+   *
+   * See the definitions.test.js file for examples of how to create your own updated definitions.json.
+   *
+   * @param enums - A json encoding of the core types, transaction types, transaction results, transaction names, and fields.
+   * @param types - A list of type objects with the same name as the fields defined.
+   *              You can use the coreTypes object if you are not adding new types.
+   */
+  constructor(
```

Comment:
This class should have an alternate constructor that takes in a file location and converts it into a `DefinitionsContents`

Reply:
Chatted with @ckniffen about this and decided that having us import the json directly seems to add a lot of complexity without much value add. Importing a json file seems like a fine thing to ask the user to do as an input.